File: test-suite/src/main/java/io/micronaut/gcp/pubsub/subscriber/MessageProcessor.java
package io.micronaut.gcp.pubsub.subscriber;

import com.google.pubsub.v1.PubsubMessage;
import io.micronaut.gcp.pubsub.support.Animal;
import reactor.core.publisher.Mono;

public interface MessageProcessor {
    default Mono<Boolean> handleByteArrayMessage(byte[] message) {
        return Mono.just(Boolean.TRUE);
    }

    default Mono<Boolean> handlePubSubMessage(PubsubMessage pubsubMessage) {
        return Mono.just(Boolean.TRUE);
    }

    default Mono<Boolean> handleAnimalMessage(Animal message) {
        return Mono.just(Boolean.TRUE);
    }
}

File: test-suite-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/subscriber/MessageProcessor.kt
package io.micronaut.gcp.pubsub.subscriber

import com.google.pubsub.v1.PubsubMessage
import io.micronaut.gcp.pubsub.support.Animal
import io.micronaut.messaging.Acknowledgement
import reactor.core.publisher.Mono

open class MessageProcessor {

    open fun handleByteArrayMessage(message: ByteArray) = Mono.just(java.lang.Boolean.TRUE)

    open fun handlePubsubMessage(pubsubMessage: PubsubMessage) = Mono.just(java.lang.Boolean.TRUE)

    open fun handleAnimalMessage(message: Animal) = Mono.just(java.lang.Boolean.TRUE)

    open fun recordAcknowledgement(acknowledgement: Acknowledgement) {}
}
